test/gas/EndToEnd.t.sol

```diff
@@ -0,0 +1,103 @@
+        usdc.mint(address(account), 10000e6);
+        usdc.mint(eoaUser, 10000e6);
+        
+        target = new MockTarget();
+    }
+    
+    // Native ETH transfer comparison between ERC-4337 and EOA
+    function test_transfer_native() public {
+        userOpCalldata = abi.encodeCall(CoinbaseSmartWallet.execute, (address(0x1234), 1 ether, ""));
+        UserOperation memory op = _getUserOpWithSignature();
+        
+        bytes memory handleOpsCalldata = abi.encodeCall(entryPoint.handleOps, (_makeOpsArray(op), payable(bundler)));
+        console2.log("test_transfer_native ERC-4337 calldata size:", handleOpsCalldata.length);
+        
+        vm.startSnapshotGas("e2e_transfer_native_4337");
+        _sendUserOperation(op);
+        uint256 gas4337 = vm.stopSnapshotGas();
+        console2.log("test_transfer_native ERC-4337 gas:", gas4337);
+        
+        console2.log("test_transfer_native EOA calldata size:", uint256(0));
+        
+        vm.prank(eoaUser);
+        vm.startSnapshotGas("e2e_transfer_native_eoa");
+        payable(address(0x1234)).transfer(1 ether);
+        uint256 gasEOA = vm.stopSnapshotGas();
+        console2.log("test_transfer_native EOA gas:", gasEOA);
+        console2.log("Gas overhead (4337/EOA):", (gas4337 * 100) / gasEOA, "%");
+    }
+    
+    // ERC20 transfer comparison between ERC-4337 and EOA
+    function test_transfer_erc20() public {
+        userOpCalldata = abi.encodeCall(
+            CoinbaseSmartWallet.execute,
+            (address(usdc), 0, abi.encodeCall(usdc.transfer, (address(0x5678), 100e6)))
+        );
+        UserOperation memory op = _getUserOpWithSignature();
+        
+        bytes memory handleOpsCalldata = abi.encodeCall(entryPoint.handleOps, (_makeOpsArray(op), payable(bundler)));
+        console2.log("test_transfer_erc20 ERC-4337 calldata size:", handleOpsCalldata.length);
+        
+        vm.startSnapshotGas("e2e_transfer_erc20_4337");
+        _sendUserOperation(op);
+        uint256 gas4337 = vm.stopSnapshotGas();
+        console2.log("test_transfer_erc20 ERC-4337 gas:", gas4337);
+        
+        bytes memory eoaCalldata = abi.encodeCall(usdc.transfer, (address(0x5678), 100e6));
+        console2.log("test_transfer_erc20 EOA calldata size:", eoaCalldata.length);
+        
+        vm.prank(eoaUser);
+        vm.startSnapshotGas("e2e_transfer_erc20_eoa");
+        usdc.transfer(address(0x5678), 100e6);
+        uint256 gasEOA = vm.stopSnapshotGas();
+        console2.log("test_transfer_erc20 EOA gas:", gasEOA);
+        console2.log("Gas overhead (4337/EOA):", (gas4337 * 100) / gasEOA, "%");
+    }
+    
+    function _makeOpsArray(UserOperation memory op) internal pure returns (UserOperation[] memory) {
+        UserOperation[] memory ops = new UserOperation[](1);
+        ops[0] = op;
+        return ops;
+    }
+    
+    function _sign(UserOperation memory userOp) internal view override returns (bytes memory signature) {
```

Comment:
Not seeing where this is being used. Does the parent test contract require use to override this?